libminifi/src/io/tls/TLSSocket.cpp

```diff
@@ -170,22 +174,19 @@ void TLSSocket::closeStream() {
  * @param listeners number of listeners in the queue
  */
 TLSSocket::TLSSocket(const std::shared_ptr<TLSContext> &context, const std::string &hostname, const uint16_t port, const uint16_t listeners)
-    : Socket(context, hostname, port, listeners),
-      ssl_(0) {
+    : Socket(context, hostname, port, listeners) {
   logger_ = logging::LoggerFactory<TLSSocket>::getLogger();
   context_ = context;
 }
 
 TLSSocket::TLSSocket(const std::shared_ptr<TLSContext> &context, const std::string &hostname, const uint16_t port)
-    : Socket(context, hostname, port, 0),
-      ssl_(0) {
+    : Socket(context, hostname, port, 0) {
   logger_ = logging::LoggerFactory<TLSSocket>::getLogger();
   context_ = context;
 }
 
-TLSSocket::TLSSocket(const TLSSocket &&d)
-    : Socket(std::move(d)),
-      ssl_(0) {
+TLSSocket::TLSSocket(TLSSocket &&d) noexcept
```

Comment:
I think this implementation is wrong (it was wrong before this PR).
TLSSocket has a bunch of data members that are ignored by this move constructor, among them an std::map, whose move constructor is not noexcept, as far as I know, so TLSSocket's can't be either.
Not to mention that the single data member it does not ignore is copied, but since that's a shared_ptr, that part might have actually worked.

Reply:
I suggest resolving the code bugs in separate issues, since there are plenty.

Reply:
Since this is a substantial change to the sockets, I would like to run extensive tests on them on multiple platforms, which I would not like to do on code we know is faulty. This is exactly as bad as it was, but the Socket's move constructor has been changed, but it is still bad.
I am OK with a new issue or separate commits in this PR, altough I don't see the point, I think it fits in the scope of the current issue very well.
I am -1 on this PR until these are fixed.

Reply:
I've created MINIFICPP-1154 for the bugs because MINIFICPP-1119 is only about the unification of the implementations. PR renamed to include the new issue reference.